File: php/SwaggerClient-php/lib/Model/TransactionItem.php
<?php

namespace Swagger\Client\Model;

use \ArrayAccess;
use \Swagger\Client\ObjectSerializer;

class TransactionItem implements ModelInterface, ArrayAccess
{
    const DISCRIMINATOR = null;

    protected static $swaggerModelName = 'TransactionItem';

    protected static $swaggerTypes = [
        'tax_included' => 'bool',
        'credit' => 'bool',
        'extended_amount' => 'double',
        'item_number' => 'string',
        'unit_cost' => 'double',
        'quantity' => 'int',
        'description' => 'string',
        'discount' => 'double',
        'tax' => 'double',
        'commodity_code' => 'string',
        'unit_of_measure' => 'string'
    ];

    protected static $swaggerFormats = [
        'tax_included' => null,
        'credit' => null,
        'extended_amount' => 'double',
        'item_number' => null,
        'unit_cost' => 'double',
        'quantity' => 'int32',
        'description' => null,
        'discount' => 'double',
        'tax' => 'double',
        'commodity_code' => null,
        'unit_of_measure' => null
    ];

    public static function swaggerTypes()
    {
        return self::$swaggerTypes;
    }

    public static function swaggerFormats()
    {
        return self::$swaggerFormats;
    }

    protected static $attributeMap = [
        'tax_included' => 'TaxIncluded',
        'credit' => 'Credit',
        'extended_amount' => 'ExtendedAmount',
        'item_number' => 'ItemNumber',
        'unit_cost' => 'UnitCost',
        'quantity' => 'Quantity',
        'description' => 'Description',
        'discount' => 'Discount',
        'tax' => 'Tax',
        'commodity_code' => 'CommodityCode',
        'unit_of_measure' => 'UnitOfMeasure'
    ];

    protected static $setters = [
        'tax_included' => 'setTaxIncluded',
        'credit' => 'setCredit',
        'extended_amount' => 'setExtendedAmount',
        'item_number' => 'setItemNumber',
        'unit_cost' => 'setUnitCost',
        'quantity' => 'setQuantity',
        'description' => 'setDescription',
        'discount' => 'setDiscount',
        'tax' => 'setTax',
        'commodity_code' => 'setCommodityCode',
        'unit_of_measure' => 'setUnitOfMeasure'
    ];

    protected static $getters = [
        'tax_included' => 'getTaxIncluded',
        'credit' => 'getCredit',
        'extended_amount' => 'getExtendedAmount',
        'item_number' => 'getItemNumber',
        'unit_cost' => 'getUnitCost',
        'quantity' => 'getQuantity',
        'description' => 'getDescription',
        'discount' => 'getDiscount',
        'tax' => 'getTax',
        'commodity_code' => 'getCommodityCode',
        'unit_of_measure' => 'getUnitOfMeasure'
    ];

    public static function attributeMap()
    {
        return self::$attributeMap;
    }

    public static function setters()
    {
        return self::$setters;
    }

    public static function getters()
    {
        return self::$getters;
    }

    public function getModelName()
    {
        return self::$swaggerModelName;
    }

    protected $container = [];

    public function __construct(array $data = null)
    {
        $this->container['tax_included'] = isset($data['tax_included']) ? $data['tax_included'] : null;
        $this->container['credit'] = isset($data['credit']) ? $data['credit'] : null;
        $this->container['extended_amount'] = isset($data['extended_amount']) ? $data['extended_amount'] : null;
        $this->container['item_number'] = isset($data['item_number']) ? $data['item_number'] : null;
        $this->container['unit_cost'] = isset($data['unit_cost']) ? $data['unit_cost'] : null;
        $this->container['quantity'] = isset($data['quantity']) ? $data['quantity'] : null;
        $this->container['description'] = isset($data['description']) ? $data['description'] : null;
        $this->container['discount'] = isset($data['discount']) ? $data['discount'] : null;
        $this->container['tax'] = isset($data['tax']) ? $data['tax'] : null;
        $this->container['commodity_code'] = isset($data['commodity_code']) ? $data['commodity_code'] : null;
        $this->container['unit_of_measure'] = isset($data['unit_of_measure']) ? $data['unit_of_measure'] : null;
    }

    public function listInvalidProperties()
    {
        $invalidProperties = [];

        return $invalidProperties;
    }

    public function valid()
    {
        return count($this->listInvalidProperties()) === 0;
    }

    public function getTaxIncluded()
    {
        return $this->container['tax_included'];
    }

    public function setTaxIncluded($tax_included)
    {
        $this->container['tax_included'] = $tax_included;

        return $this;
    }

    public function getCredit()
    {
        return $this->container['credit'];
    }

    public function setCredit($credit)
    {
        $this->container['credit'] = $credit;

        return $this;
    }

    public function getExtendedAmount()
    {
        return $this->container['extended_amount'];
    }

    public function setExtendedAmount($extended_amount)
    {
        $this->container['extended_amount'] = $extended_amount;

        return $this;
    }

    public function getItemNumber()
    {
        return $this->container['item_number'];
    }

    public function setItemNumber($item_number)
    {
        $this->container['item_number'] = $item_number;

        return $this;
    }

    public function getUnitCost()
    {
        return $this->container['unit_cost'];
    }

    public function setUnitCost($unit_cost)
    {
        $this->container['unit_cost'] = $unit_cost;

        return $this;
    }

    public function getQuantity()
    {
        return $this->container['quantity'];
    }

    public function setQuantity($quantity)
    {
        $this->container['quantity'] = $quantity;

        return $this;
    }

    public function getDescription()
    {
        return $this->container['description'];
    }

    public function setDescription($description)
    {
        $this->container['description'] = $description;

        return $this;
    }

    public function getDiscount()
    {
        return $this->container['discount'];
    }

    public function setDiscount($discount)
    {
        $this->container['discount'] = $discount;

        return $this;
    }

    public function getTax()
    {
        return $this->container['tax'];
    }

    public function setTax($tax)
    {
        $this->container['tax'] = $tax;

        return $this;
    }

    public function getCommodityCode()
    {
        return $this->container['commodity_code'];
    }

    public function setCommodityCode($commodity_code)
    {
        $this->container['commodity_code'] = $commodity_code;

        return $this;
    }

    public function getUnitOfMeasure()
    {
        return $this->container['unit_of_measure'];
    }

    public function setUnitOfMeasure($unit_of_measure)
    {
        $this->container['unit_of_measure'] = $unit_of_measure;

        return $this;
    }
    public function offsetExists($offset)
    {
        return isset($this->container[$offset]);
    }

    public function offsetGet($offset)
    {
        return isset($this->container[$offset]) ? $this->container[$offset] : null;
    }

    public function offsetSet($offset, $value)
    {
        if (is_null($offset)) {
            $this->container[] = $value;
        } else {
            $this->container[$offset] = $value;
        }
    }

    public function offsetUnset($offset)
    {
        unset($this->container[$offset]);
    }

    public function __toString()
    {
        if (defined('JSON_PRETTY_PRINT')) { // use JSON pretty print
            return json_encode(
                ObjectSerializer::sanitizeForSerialization($this),
                JSON_PRETTY_PRINT
            );
        }

        return json_encode(ObjectSerializer::sanitizeForSerialization($this));
    }
}

File: CodeSamples/python/swagger_client/models/transaction_item.py
# coding: utf-8

class TransactionItem(object):
    swagger_types = {
        'tax_included': 'bool',
        'credit': 'bool',
        'extended_amount': 'float',
        'item_number': 'str',
        'unit_cost': 'float',
        'quantity': 'int',
        'description': 'str',
        'discount': 'float',
        'tax': 'float',
        'commodity_code': 'str',
        'unit_of_measure': 'str'
    }

    attribute_map = {
        'tax_included': 'TaxIncluded',
        'credit': 'Credit',
        'extended_amount': 'ExtendedAmount',
        'item_number': 'ItemNumber',
        'unit_cost': 'UnitCost',
        'quantity': 'Quantity',
        'description': 'Description',
        'discount': 'Discount',
        'tax': 'Tax',
        'commodity_code': 'CommodityCode',
        'unit_of_measure': 'UnitOfMeasure'
    }

    def __init__(self, tax_included=None, credit=None, extended_amount=None, item_number=None, unit_cost=None, quantity=None, description=None, discount=None, tax=None, commodity_code=None, unit_of_measure=None):  # noqa: E501
        """TransactionItem - a model defined in Swagger"""  # noqa: E501
        self._tax_included = None
        self._credit = None
        self._extended_amount = None
        self._item_number = None
        self._unit_cost = None
        self._quantity = None
        self._description = None
        self._discount = None
        self._tax = None
        self._commodity_code = None
        self._unit_of_measure = None
        self.discriminator = None
        if tax_included is not None:
            self.tax_included = tax_included
        if credit is not None:
            self.credit = credit
        if extended_amount is not None:
            self.extended_amount = extended_amount
        if item_number is not None:
            self.item_number = item_number
        if unit_cost is not None:
            self.unit_cost = unit_cost
        if quantity is not None:
            self.quantity = quantity
        if description is not None:
            self.description = description
        if discount is not None:
            self.discount = discount
        if tax is not None:
            self.tax = tax
        if commodity_code is not None:
            self.commodity_code = commodity_code
        if unit_of_measure is not None:
            self.unit_of_measure = unit_of_measure

    @property
    def tax_included(self):
        return self._tax_included

    @tax_included.setter
    def tax_included(self, tax_included):
        self._tax_included = tax_included

    @property
    def credit(self):
        return self._credit

    @credit.setter
    def credit(self, credit):
        self._credit = credit

    @property
    def extended_amount(self):
        return self._extended_amount

    @extended_amount.setter
    def extended_amount(self, extended_amount):
        self._extended_amount = extended_amount

    @property
    def item_number(self):
        return self._item_number

    @item_number.setter
    def item_number(self, item_number):
        self._item_number = item_number

    @property
    def unit_cost(self):
        return self._unit_cost

    @unit_cost.setter
    def unit_cost(self, unit_cost):
        self._unit_cost = unit_cost

    @property
    def quantity(self):
        return self._quantity

    @quantity.setter
    def quantity(self, quantity):
        self._quantity = quantity

    @property
    def description(self):
        return self._description

    @description.setter
    def description(self, description):
        self._description = description

    @property
    def discount(self):
        return self._discount

    @discount.setter
    def discount(self, discount):
        self._discount = discount

    @property
    def tax(self):
        return self._tax

    @tax.setter
    def tax(self, tax):
        self._tax = tax

    @property
    def commodity_code(self):
        return self._commodity_code

    @commodity_code.setter
    def commodity_code(self, commodity_code):
        self._commodity_code = commodity_code

    @property
    def unit_of_measure(self):
        return self._unit_of_measure

    @unit_of_measure.setter
    def unit_of_measure(self, unit_of_measure):
        self._unit_of_measure = unit_of_measure
